pkg/ingester/metrics.go

```diff
@@ -25,6 +25,7 @@ type ingesterMetrics struct {
 	queriedSamples        prometheus.Histogram
 	queriedSeries         prometheus.Histogram
 	queriedChunks         prometheus.Histogram
+	tsdbReplayTime        prometheus.Histogram
```

Comment:
I think this should go to `TSDBState` struct in ingester_v2.go. Also technically, it's "open" time for existing DBs, so perhaps "tsdbStartupOpenTime" (?)

Reply:
Yea that's fair, I forgot we had metrics in that struct.

Reply:
Well, actually, should we have metrics in there? Or should we move all metrics into the metrics struct?

Reply:
I think blocks-specific metrics should go there. Chunks and shared metrics between chunks and blocks go to metrics.go

Reply:
I get that delineation, but from another perspective it feels strange to have a metrics file that doesn't contain all the metrics.

Reply:
My feedback on the topic:

1. `tsdbMetrics struct` is a different thing and it's used for the Thanos mapped metrics. I believe this metric should go to `TSDBState struct` where we already have other blocks-storage related metrics tracked in Cortex (ie. `compactionsTriggered`).
2. The `tsdb` prefix in the `tsdbReplayTime` is redundant and should be removed. We could call it `walReplayDuration` which is that it is (I understand Peter's feedback that technically it's TSDB open, but practically the time there is taken by WAL replay and we track this metric only for existing TSDBs and not new ones we just open)

Then, we can discuss whether it's a good or bad things having ingester metrics defined in multiple places, but this shouldn't be address by this PR (we can discuss it offline and eventually come up with a dedicated PR to address it).

Reply:
Other thoughts. We already have `cortex_ingester_wal_replay_duration_seconds` metric (chunks storage). Why don't we reuse the same?

Reply:
Only problem with that is that's a gauge and not a histogram.